fix search icon contrast issues

yiningwang11 opened the pull request:
fix #660 
I modified the theme colour of the search icon, now it looks like this:
|          | light mode  | dark mode |
| ------------- | ------------- | ------------- |
unhovered | <img width="859" alt="image" src="https://user-images.githubusercontent.com/108029327/234316510-d039f627-e5ed-4448-88f9-47b291b9527a.png">  | <img width="847" alt="image" src="https://user-images.githubusercontent.com/108029327/234318089-35e24c22-e89e-4c22-9d4e-cab34fcf2ce5.png">  |
hovered | <img width="832" alt="image" src="https://user-images.githubusercontent.com/108029327/234317636-7bbd38c4-48fd-478f-ba04-ffb9ec1531c9.png"> | <img width="837" alt="image" src="https://user-images.githubusercontent.com/108029327/234318265-1a4ac6d5-3985-466d-b879-68db3b74e0ab.png">

filiptronicek commented on webui/src/default/theme.tsx:
```diff
@@ -28,7 +28,7 @@ export default function createDefaultTheme(themeType: 'light' | 'dark'): Theme {
         palette: {
             primary: {
-                main: themeType === 'dark' ? '#eeeeee' : '#444',
+                main: themeType === 'dark' ? '#111111' : '#ffffff',
```
Nice one-liner! I hope this won't have unintended consequences 😄

Is there a way to maybe make a new Theme just for that one search component? I think if we changed it here we'd have to test the whole Web UI for color changes and side-effects.

yiningwang11 replied:
Thanks for the comment!
Let me see if there's a way to separate the theme for this component. I'm thinking maybe I should create a new theme and apply it to the search icon button individually?

filiptronicek replied:
I think that would be the better option here indeed. Looping in @amvanbaren, maybe he has some ideas :)

amvanbaren replied:
I see the main issue has already been fixed. 
There are 2 more things I noticed:
- You don't have to pass `themeType` through props. You can use `const { pageSettings } = useContext(MainContext);` instead. For example: https://github.com/eclipse/openvsx/blob/5df9b741623a7fcee303e8a241067b54becb2110/webui/src/pages/admin-dashboard/namespace-admin.tsx#L21
- You can define style classes instead of using inline styles, e.g. iconButtonLight and iconButtonDark.
https://github.com/eclipse/openvsx/blob/5df9b741623a7fcee303e8a241067b54becb2110/webui/src/pages/extension-list/extension-list-searchfield.tsx#L29-L37
And then you can use it like this:
https://github.com/eclipse/openvsx/blob/5df9b741623a7fcee303e8a241067b54becb2110/webui/src/pages/extension-detail/extension-detail.tsx#L348-L352

amvanbaren reviewed (commented):
Looks way better! One nitpick: you can replace 
```
const prefersDarkMode = pageSettings?.themeType === 'dark' ? true : false;
``` 
with 
```
const searchIconClass = pageSettings?.themeType === 'dark' ? classes.searchIconDark : classes.searchIconLight;
``` 
and then use that to style the `SearchIcon`, so that you only have one ternary (`condition ? x : y`) statement. 

@filiptronicek Does it fix the contrast issues?

filiptronicek commented:
The contrast issues are solved.

amvanbaren commented:
Ok, LGTM. @yiningwang11 can you squash your commits, so that this PR gets merged into master as 1 commit?

yiningwang11 commented:
Hello @amvanbaren ,
I am following this [guide](https://www.freecodecamp.org/news/git-squash-commits/#:~:text=So%20if%20you%20want%20to,to%20go%20back%206%20commits.&text=Now%2C%20you%20need%20to%20replace,and%20combine%20them%20all%20together.) here to do the squash.
I did `git stash` and then `git rebase -i HEAD~6`. Then I changed the `pick` to `squash` as suggested. After modifying the commit message I see a message on the terminal as:
```
[detached HEAD 3caf568] Ready to deploy
 Author: amvanbaren <aart.vanbaren@eclipse-foundation.org>
 Date: Tue Apr 18 13:42:54 2023 +0300
 5 files changed, 123 insertions(+), 77 deletions(-)
 create mode 100644 server/src/main/java/org/eclipse/openvsx/migration/GenerateKeyPairJobService.java
Successfully rebased and updated refs/heads/contrastIcon.
```

I noticed that there must be something wrong with my process because the commits on this branch are not being squashed and the message I got above is clearly not what I expected.
I suspect this is because one of my commits `e6a1de6` is still going through the checks. 
I'm kindly asking if there is a way to fix or undo what I have done. Apologies for the mess that I brought :(

amvanbaren commented:
@yiningwang11 Luckily you haven't pushed `contrastIcon` yet. You can reset your local branch to reflect the remote one: https://www.freecodecamp.org/news/git-reset-origin-how-to-reset-a-local-branch-to-remote-tracking-branch/#reset.

Then you should rebase the last 5 commits (`git rebase -i HEAD~5`):
- `pick` cd6bb87 
- `squash` 8c19e96, 0696508, 4f0eeaf and e6a1de6. 

You can see your list of commits here: https://github.com/eclipse/openvsx/pull/730/commits.
The changes of the resulting (squashed) commit should be the same as https://github.com/eclipse/openvsx/pull/730/files.

yiningwang11 commented:
Hello @amvanbaren ,
Thank you for your help!
Following the guide, I have reset my local branch `contrastIcon` by doing `git reset --hard origin/contrastIcon` to reflect the remote one and it now looks good. I am now able to proceed to do the rebase for squash.

filiptronicek reviewed (approved):
Awesome stuff!

I think the next nice step is to maybe add a transition to the button, but maybe that's for another PR :)
![Recording 2023-05-11 at 19 36 28](https://github.com/eclipse/openvsx/assets/29888641/917910c9-b87c-47a1-ae05-57b95664e74e)
